VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+                    Write-Output "Workflow Completed with an unrecognized status of $($response.status). Please check before proceeding."
+                }
+                Write-Output "Certificates for Components Associated with Domain $($workloadDomain) Generated"
+        } else {
+            Write-Error "cannot authenticate to SDDC Manager"
+        }
+    } else {
+        Write-Error "cannot connect to SDDC Manager"
+    }
+}
+
+Function Invoke-VCFInstallCertificates {
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $workloadDomain
+    )
+    if (Test-VCFConnection -server $server) {
+        if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+            $domainType = Get-VCFWorkloadDomain -name $workloadDomain
+            $resourcesObject = gatherSddcInventory -domainType $domainType.type -workloadDomain $workloadDomain
+            Write-Output "Populating $($workloadDomain)-updateCertificateSpec.json"
+            $path = Join-Path $pwd ""
+            $operationTypeJson = '{
+                "operationType": "INSTALL",
+                '
+              $resourcesBodyObject += [pscustomobject]@{
+                  resources = $resourcesObject
+              }
+              $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $path"temp.json"
+              Get-Content $path"temp.json" | Select-Object -Skip 1 | Set-Content $path"temp1.json"
+              $resouresJson = Get-Content $path"temp1.json" -Raw
+              Remove-Item -Path $path"temp.json"
+              Remove-Item -Path $path"temp1.json"
+              $requestCertificateSpecJson = $operationTypeJson + $resouresJson
+              $requestCertificateSpecJson | Out-File $path"$($workloadDomain)-updateCertificateSpec.json"
+
+              # Install Certificate
+              Try {
+                Write-Output "Installing Signed Certificates for Components Associated with Domain $($workloadDomain) (can take 60+ mins)"
+                $myTaskId = Set-VCFCertificate -domainName $($workloadDomain) -json $path"$($workloadDomain)-updateCertificateSpec.json"
+                $pollLoopCounter = 0
+                Do {
+                    If ($pollLoopCounter % 10 -eq 0) {
+                        Write-Output "Checking Installation of Signed Certificates for Components Associated with Domain $($workloadDomain)"
+                    }
+                    $response = Get-VCFTask $myTaskId.id
+                    If ($response.status -in "In Progress","IN_PROGRESS") {
+                        If (($pollLoopCounter % 10 -eq 0) -AND ($pollLoopCounter -gt 9)) {
+                            Write-Output "Certificate installation still in progress"
```

Comment:
```suggestion
                            Write-Output "Installation of signed certificates is still in progress for workload domain ($($workloadDomain))..."
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)